<launch>
    <arg name="planner_name"              default="MCAstar"/>                   <!-- 选用的规划器种类，大小写敏感。 MCAstar / Astar -->
    <arg name="input_map_topic"           default="/grid_cost_map/global_occupancy_grid_map"/>  <!-- 输入的代价地图话题 -->
    <arg name="input_goal_topic"          default="/move_base_simple/goal"/>    <!-- 输入的终点话题 -->
    <arg name="output_extended_map_topic" default="extended_map"/>              <!-- 输出的膨胀后的地图 -->
    <arg name="output_path_topic"         default="path"/>                      <!-- 输出的原始路径，未经过平滑等操作，是节点的路径 -->
    <arg name="output_smooth_path_topic"  default="smooth_path"/>               <!-- 输出的平滑后的路径，包括平滑、冗余点去除、降采样操作 -->

    <node pkg="global_planning" name="global_planning" type="global_planning" output="screen" >
        <!-- MCAstar规划器对应的参数 -->
        <rosparam command="load" file="$(find global_planning)/params/MCAstar_params.yaml" />
        <!-- Astar规划器对应的参数 -->
        <!-- <rosparam command="load" file="$(find global_planning)/params/astar_params.yaml" /> -->

        <param name="planner_name"              type="str" value="$(arg planner_name)"/>
        <param name="input_map_topic"           type="str" value="$(arg input_map_topic)"/>
        <param name="input_goal_topic"          type="str" value="$(arg input_goal_topic)"/>
        <param name="output_extended_map_topic" type="str" value="$(arg output_extended_map_topic)"/>
        <param name="output_path_topic"         type="str" value="$(arg output_path_topic)"/>
        <param name="output_smooth_path_topic"  type="str" value="$(arg output_smooth_path_topic)"/>
    </node> 

</launch>